SQL/0000-00-00-schema.sql

```diff
@@ -2362,6 +2363,87 @@ CREATE TABLE `user_psc_rel` (
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
 INSERT INTO user_psc_rel (UserID, CenterID) SELECT 1, CenterID FROM psc;
 
+-- Associates modules with the service available for each
+DROP TABLE IF EXISTS `notification_modules`;
+CREATE TABLE `notification_modules` (
+      `id` int(10) unsigned auto_increment NOT NULL,
+      `module_name` varchar(100) NOT NULL,
+      `operation_type` varchar(100) NOT NULL,
+      `as_admin` enum('Y','N') NOT NULL DEFAULT 'N',
+      `template_file` varchar(100) NOT NULL,
+      `description` varchar(255) DEFAULT NULL,
+      PRIMARY KEY (`id`),
+      KEY (`module_name`),
+      UNIQUE(module_name,operation_type)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+-- Associates modules with the service available for each
+DROP TABLE IF EXISTS `notification_services`;
+CREATE TABLE `notification_services` (
+      `id` int(10) unsigned auto_increment NOT NULL,
+      `service` VARCHAR(50) NOT NULL,
+      PRIMARY KEY (`id`),
+      UNIQUE(service)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+-- saves users preferences for notification type
+DROP TABLE IF EXISTS `notification_modules_services_rel`;
+CREATE TABLE `notification_modules_services_rel` (
+      `module_id` int(10) unsigned NOT NULL,
+      `service_id` int(10) unsigned NOT NULL,
+      PRIMARY KEY (`module_id`,`service_id`),
+      CONSTRAINT `FK_notification_modules_services_rel_1` FOREIGN KEY (`module_id`) REFERENCES `notification_modules` (`id`),
+      CONSTRAINT `FK_notification_modules_services_rel_2` FOREIGN KEY (`service_id`) REFERENCES `notification_services` (`id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+-- saves users preferences for notification type
+DROP TABLE IF EXISTS `users_notifications_rel`;
+CREATE TABLE `users_notifications_rel` (
+      `user_id` int(10) unsigned NOT NULL,
+      `module_id` int(10) unsigned NOT NULL,
+      `service_id` int(10) unsigned NOT NULL,
+      PRIMARY KEY (`user_id`,`module_id`,`service_id`),
+      CONSTRAINT `FK_notifications_users_rel_1` FOREIGN KEY (`user_id`) REFERENCES `users` (`ID`),
+      CONSTRAINT `FK_notifications_users_rel_2` FOREIGN KEY (`module_id`) REFERENCES `notification_modules` (`id`),
+      CONSTRAINT `FK_notifications_users_rel_3` FOREIGN KEY (`service_id`) REFERENCES `notification_services` (`id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+-- history log
+DROP TABLE IF EXISTS `notification_history`;
+CREATE TABLE `notification_history` (
+      `id` int(10) unsigned auto_increment NOT NULL,
+      `module_id` int(10) unsigned NOT NULL,
+      `service_id` int(10) unsigned NOT NULL,
+      `date_sent` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+      `trigger_user` int(10) unsigned NOT NULL,
+      `target_user` int(10) unsigned NOT NULL,
+      PRIMARY KEY (`id`),
+--      CONSTRAINT `FK_notification_history_1` FOREIGN KEY (`module_id`) REFERENCES `notification_modules` (`id`),
```

Comment:
the question is, should history tables have foreign keys at all ? what if you decide to remove a service or remove a module. should the history table prevent you ? should we delete from the history the entries referencing those fields ?

thats a bigger question than notifications obviously